### PR TITLE
add multi-stage taxi trip to SaoPauloTaxiPredictor

### PR DESCRIPTION
**Pull request by MayuriAnn**
As suggested in https://github.com/eqasim-org/eqasim-java/issues/191#issuecomment-1950041727, I followed the code on https://github.com/eqasim-org/eqasim-java/blob/a8a73b53dbfc085e6eabddcd8ba43fd72013b063/core/src/main/java/org/eqasim/core/simulation/mode_choice/utilities/predictors/CarPredictor.java#L36-L45 to update the code of SaoPauloTaxiPredictor class.

### CODE REVIEW

File: sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java [hunk 1]
```diff
@@ -3,39 +3,55 @@
...
 		this.costModel = costModel;
+		this.parameters = parameters;
 	}
 
 	@Override
 	public TaxiVariables predict(Person person, DiscreteModeChoiceTrip trip, List<? extends PlanElement> elements) {
-		if (elements.size() > 1) {
-			throw new IllegalStateException("We do not support multi-stage taxi trips yet.");
+		
+		double taxiTravelTime_min = parameters.car.constantParkingSearchPenalty_min;
```
Comment (balacmi) [inline sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java:34]: Can you please remove this line as it is only used for car trips and for for taxis.

File: sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java [hunk 2]
```diff
@@ -3,39 +3,55 @@
...
+		double accessEgressTime_min = parameters.car.additionalAccessEgressWalkTime_min;
```
Comment (balacmi) [inline sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java:35]: Same as above

File: sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java [hunk 3]
```diff
@@ -3,39 +3,55 @@
...
 
 public class SaoPauloTaxiPredictor extends CachedVariablePredictor<TaxiVariables> {
 	private final CostModel costModel;
+	private final ModeParameters parameters;
 
 	@Inject
-	public SaoPauloTaxiPredictor(@Named("taxi") CostModel costModel) {
+	public SaoPauloTaxiPredictor(ModeParameters parameters, @Named("taxi") CostModel costModel) {
 		this.costModel = costModel;
+		this.parameters = parameters;
```
Comment (balacmi) [inline sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java:28]: ModeParameters object is for now not used, please remove it (see below my further comments).

**Review by balacmi — CHANGES_REQUESTED** [review]
Thank you for looking into this. Can you please address my above comments and I will merge the PR.

**Comment by MayuriAnn** [comment]
Please check if its correct now.

**Review by balacmi — APPROVED** [review]
Looks good, thank you.

**Comment by balacmi** [comment]
May I also ask you to add to the `CHANGELOG.md` file the update you made, so we can keep track of what changed between releases.

**Review by balacmi — APPROVED** [review]
Great

**Pull request merged**